src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs

```diff
@@ -311,6 +269,55 @@ private bool IsTypeExcludedFromValidation(IList<IExcludeTypeValidationFilter> fi
             return filters.Any(filter => filter.IsTypeExcluded(type));
         }
 
+        private IList<IModelValidator> ExpandValidationNodeAndGetValidators(
+            ValidationContext context, ModelExplorer modelExplorer)
+        {
+            var validationNode = context.ValidationNode;
+            if (validationNode == null || !validationNode.BuildChildNodesUsingModel || validationNode.Model == null)
```

Comment:
there should be another condition here, we should return if there are already child nodes present in the validation node. That would mean that some nodes have been created due to model binding, and we should honor those. (TryUpdateModel etc).